model/labels/regexp.go

```diff
@@ -856,6 +884,8 @@ func optimizeEqualStringMatchers(input StringMatcher, threshold int) StringMatch
 		caseSensitive    bool
 		caseSensitiveSet bool
 		numValues        int
+		numPrefixes      int
+		minPrefixLength  = math.MaxInt
 	)
 
 	// Analyse the input StringMatcher to count the number of occurrences
```

Comment:
[nit] I would rename `analyseCallback` to something like `analyseEqualMatcherCallback`